packages/gestalt/src/Fieldset.js

```diff
@@ -0,0 +1,34 @@
+// @flow strict
+import type { Node } from 'react';
+import cx from 'classnames';
+import Text from './Text.js';
+import labelStyles from './Label.css';
+import formStyles from './FormElement.css';
+import formLabelStyles from './FormLabel.css';
+import boxWhitespaceStyles from './boxWhitespace.css';
+import whitespaceStyles from './Whitespace.css';
+import boxStyles from './Box.css';
+
+type Props = {|
+  children: Node,
+  legend: string,
+  legendDisplay?: 'visible' | 'hidden',
+|};
+
+export default function Fieldset({ legend, legendDisplay = 'visible', children }: Props): Node {
+  return (
+    <fieldset className={cx(formStyles.unstyled, whitespaceStyles.p0, whitespaceStyles.m0)}>
+      <legend
+        className={cx(
+          labelStyles.label,
+          formLabelStyles.formLabel,
+          boxWhitespaceStyles.paddingX0,
```

Comment:
I was like why do we need this  `boxWhitespaceStyles.paddingX0,`

It seems leyend come with padding. 
![image](https://user-images.githubusercontent.com/10593890/117062343-55b5b680-acf1-11eb-867d-3ed03f22cef3.png)